ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java

```diff
@@ -154,12 +162,58 @@ public List<DDLInfo> getTableOperations(AmoroTable<?> amoroTable) {
 
   @Override
   public List<PartitionBaseInfo> getTablePartition(AmoroTable<?> amoroTable) {
-    throw new UnsupportedOperationException();
+    FileStoreTable table = getTable(amoroTable);
+    AbstractFileStore<?> store = (AbstractFileStore<?>) table.store();
+    FileStorePathFactory fileStorePathFactory = store.pathFactory();
+    List<ManifestEntry> files = store.newScan().plan().files(FileKind.ADD);
+    Map<BinaryRow, Map<Integer, List<DataFileMeta>>> groupByPartFiles = groupByPartFiles(files);
+
+    List<PartitionBaseInfo> partitionBaseInfoList = new ArrayList<>();
+    for (Map.Entry<BinaryRow, Map<Integer, List<DataFileMeta>>> groupByPartitionEntry : groupByPartFiles.entrySet()) {
+      for (Map.Entry<Integer, List<DataFileMeta>> groupByBucketEntry : groupByPartitionEntry.getValue().entrySet()) {
+        String partitionSt = partitionString(
+            groupByPartitionEntry.getKey(),
+            groupByBucketEntry.getKey(),
+            fileStorePathFactory);
+        int fileCount = 0;
+        int fileSize = 0;
+        long lastCommitTime = 0;
+        for (DataFileMeta dataFileMeta : groupByBucketEntry.getValue()) {
+          fileCount++;
+          fileSize += dataFileMeta.fileSize();
+          lastCommitTime = Math.max(lastCommitTime, dataFileMeta.creationTime().getMillisecond());
+        }
+        partitionBaseInfoList.add(new PartitionBaseInfo(partitionSt, fileCount, fileSize, lastCommitTime));
+      }
+    }
+    return partitionBaseInfoList;
   }
 
   @Override
   public List<PartitionFileBaseInfo> getTableFile(AmoroTable<?> amoroTable, String partition) {
```

Comment:
`getTableFiles` may be a better method name.

Reply:
BTW, we lost some `@Override` for methods in `com.netease.arctic.server.dashboard.MixedAndIcebergTableDescriptor`.

Reply:
yeah